execution_engine/src/storage/global_state/lmdb.rs

```diff
@@ -281,16 +289,58 @@ impl StateProvider for LmdbGlobalState {
         correlation_id: CorrelationId,
         trie_keys: Vec<Digest>,
     ) -> Result<Vec<Digest>, Self::Error> {
-        let txn = self.environment.create_read_txn()?;
-        let missing_descendants =
-            missing_trie_keys::<Key, StoredValue, lmdb::RoTransaction, LmdbTrieStore, Self::Error>(
+        if trie_keys
+            .iter()
+            .filter(|digest| {
+                !self
+                    .digests_without_missing_descendants
+                    .read()
+                    .unwrap()
+                    .contains(digest)
+            })
+            .count()
+            == 0
```

Comment:
Even if it's not `0` we still want to filter out the trie keys that we know don't have missing descendants, so that we call `missing_trie_keys` below only with the others.